src/demoserver.cpp

```diff
@@ -0,0 +1,296 @@
+#include "demoserver.h"
+#include "lobby.h"
+
+DemoServer::DemoServer(QObject *parent) : QObject(parent)
+{
+    timer = new QTimer(this);
+    timer->setTimerType(Qt::PreciseTimer);
+    timer->setSingleShot(true);
+
+    tcp_server = new QTcpServer(this);
+    connect(tcp_server, &QTcpServer::newConnection, this, &DemoServer::accept_connection);
+    connect(timer, &QTimer::timeout, this, &DemoServer::playback);
+}
+
+void DemoServer::start_server()
+{
+    if (server_started) return;
+    // Just use port 27088
+    // This should be user configurable
+    // Also, it runs into problems if multiple clients are running
+    // Someone can figure out a better solution to this later, before merge
+    if (!tcp_server->listen(QHostAddress::Any, 27088)) {
```

Comment:
The issues in this comment need to be addressed.

Reply:
Idk a better solution